integrations/appboy/lib/index.js

```diff
@@ -356,6 +367,14 @@ Appboy.prototype.track = function(track) {
     delete properties[key];
   }, reserved);
 
+  // Remove nested objects as Braze doesn't support nested objects in tracking calls
+  // https://segment.com/docs/destinations/braze/#track
+  each(function(value, key) {
+    if (typeof value === 'object') {
```

Comment:
`typeof null === 'object'` is true, maybe this would be better:
```
function isObject(val) {
  return val != null && typeof val === 'object' && Array.isArray(val) === false;
}
```

Reply:
maybe this instead? Arrays are also not allowed in tracking calls.
```suggestion
    if (val != null && typeof value === 'object') {
```

Reply:
A +1 for @NikoRoberts's suggestion:
```
if (val != null && typeof value === 'object') {
```